akka-actor/src/main/scala/akka/dispatch/SuspendDetector.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.dispatch
+
+import scala.concurrent.duration._
+
+import akka.actor.ActorSystem
+import akka.actor.ClassicActorSystemProvider
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.event.Logging
+
+object SuspendDetector extends ExtensionId[SuspendDetector] with ExtensionIdProvider {
+  override def get(system: ActorSystem): SuspendDetector = super.get(system)
+
+  override def get(system: ClassicActorSystemProvider): SuspendDetector = super.get(system)
+
+  override def lookup = SuspendDetector
+
+  override def createExtension(system: ExtendedActorSystem): SuspendDetector = new SuspendDetector(system)
+
+  /**
+   * Published to the ActorSystem's eventStream when the process suspension has been detected.
+   * Note that this message could be stale when it is received and additional check with
+   * [[SuspendDetected#wasSuspended]] is recommended to be sure that the suspension occurred recently.
+   */
+  final class SuspendDetected(suspendDetectedNanoTime: Long) {
+    def wasSuspended(since: FiniteDuration): Boolean =
+      (System.nanoTime() - suspendDetectedNanoTime <= since.toNanos)
+  }
+
+}
+
+class SuspendDetector(val system: ExtendedActorSystem) extends Extension {
+  import SuspendDetector.SuspendDetected
+
+  // FIXME config
+  private val tickInterval = 100.millis
```

Comment:
This might not have to be this frequently, because I think the scheduled task will trigger immediately when the process is woken up.